api/watcher/watcher.go

```diff
@@ -76,6 +82,10 @@ func (w *commonWatcher) init() {
 // tomb when an error occurs.
 func (w *commonWatcher) commonLoop() {
 	defer close(w.in)
+
+	ctx, cancel := context.WithCancel(w.tomb.Context(context.Background()))
+	defer cancel()
```

Comment:
Given this isn't used until way further down, can we move it closer to where it's used.

Reply:
Sure.